.travis.yml

```diff
@@ -57,7 +57,7 @@ script:
         if [[ $TRAVIS_RUST_VERSION == 1.22.1 ]]; then
           cargo fmt -- --write-mode=diff;
         fi &&
-        cargo test --release --verbose;
+        cargo test --release --verbose -- --test-threads=1;
```

Comment:
Hmm, why do we require it to be single-threaded? Do the tests fail if we execute them with multiple threads?

Reply:
yes they intermittently fail, which seems to be due to having the threads trying to modify the same files on the filesystem, e.g. on linux the uri registration process needs to modify the `~/.local/share/applications/mimeapps.list` and `~/.local/share/applications/maidsafe-example.desktop` files, so you see errors like the following:

`
Could not parse file "/home/bochaco/.local/share/applications/maidsafe-example.desktop": Key file contains line 'Termin %u' which is not a key-value pair, group, or comment
mv: cannot stat ‘/home/bochaco/.local/share/applications/mimeapps.list.new’: No such file or directory
`

Reply:
I'd suggest just `#[ignore]`ing these platform-specific tests instead and running them separately, just as we do in e.g. [maidsafe_utilties](https://github.com/maidsafe/maidsafe_utilities/blob/e261496ee929ca0ea0263a023cdfdb968eae4ac1/.travis.yml#L33). This way we can make them more predictable and isolated.

Reply:
not sure what you mean, they are not platform specific tests, it's the way uri registration works in linux so all tests would need to be ignored, moreover, I suspect the same issue could happen on windows with multiple threads writing the same registry key

Reply:
> it's the way uri registration works in linux so all tests would need to be ignored

Oh OK, so if the problem is with the function that's being used by all tests, then it surely makes sense.
Anyway, the test suite is not that heavy so it should be OK for now even if it runs within a single thread.

Reply:
it's not just a function being used by all tests, it's the functionality itself being tested that requires access to the same resources (i.e. files or registry keys)